### PR TITLE
fix(CD): deploy with correct images

### PR DESCRIPTION
**Pull request by juzhiyuan**
[x] broken pictures' link
[x] broken CI

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -29,8 +31,7 @@ jobs:
         cp -r apisix/doc/images ./website/static && rm -rf apisix/doc/images
         cp -r apisix/doc ./docs/apisix
         cd ./docs/apisix
-        find . -type f -name "*.md" -print0 | xargs -0 sed -i '' -e 's/\.\.\/images/https\:\/\/apisix\.apache\.org\/images/g'
-        find . -type f -name "*.md" -print0 | xargs -0 sed -i '' -e 's/\.\/images/https\:\/\/apisix\.apache\.org\/images/g'
+        find . -type f -name "*.md" -print0 | xargs -0 sed -i -e 's/(\.\.\/|\.\/)+?(\w+\/)?images/https\:\/\/apisix\.apache\.org\/images/g'
```
Comment (juzhiyuan) [inline .github/workflows/deploy.yml]: I'm using the following regex.

```(\.\.\/|\.\/)+?(\w+\/)?images```

image

The correct link is `https://apisix.apache.org/images/xxxx.png`

## Cases
- ```flow-load-plugin```
- ``````
- ``````
- ``````

Once those cases above have been resolved, then all current pictures will show normally.

Comment (juzhiyuan) [reply .github/workflows/deploy.yml]: The correct link is `https://apisix.apache.org/images/xxxx.png`

## Cases
- ```flow-load-plugin```
- ``````
- ``````
- ``````

Once those cases above have been resolved, then all current pictures will show normally.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -7,6 +7,8 @@ name: CI
 on:
   push:
     branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (juzhiyuan) [inline .github/workflows/deploy.yml]: This should be omitted before merging.

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -28,15 +30,23 @@ jobs:
...
-        find . -type f -name "*.md" -print0 | xargs -0 sed -i '' -e 's/\.\/images/https\:\/\/apisix\.apache\.org\/images/g'
+        # cd ./docs/apisix
+        # find . -type f -name "*.md" -print0 | xargs -0 sed -i -e 's/(\.\.\/|\.\/)+?(\w+\/)?images/https\:\/\/apisix\.apache\.org\/images/g'
 
     - name: Build
       run: cd website && yarn && yarn build
       
     - name: Copy files
-      run: cd website && cd build && cd apisix-website && cp ../../../README.md ./ && cp ../../../.asf.yaml ./
+      run: |
```
Comment (juzhiyuan) [inline .github/workflows/deploy.yml:36]: ugly but may work 🤔

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -28,15 +28,20 @@ jobs:
...
+        cd website/build/apisix-website
+        cp ../../../README.md ./ && cp ../../../.asf.yaml ./
+        cd apisix
+        find . -type f -name "*.html" -print0 | xargs -0 sed -i -e 's/src="\.\.\/\.\.\/doc\/images/src="https:\/\/apisix\.apache\.org\/images/g'
```
Comment (juzhiyuan) [inline .github/workflows/deploy.yml:40]: Ugly but works.

Comment (juzhiyuan) [reply .github/workflows/deploy.yml:40]: refer to https://github.com/apache/apisix-website/pull/86#discussion_r527350973

**Review by moonming — APPROVED** [review]

**Comment by juzhiyuan** [comment]
If there has one proper regex, we could fix this issue more easily.

**Review by LiteSun — APPROVED** [review]

**Pull request merged**